### PR TITLE
Update all addressBook instances to friendlyLink

### LINKED ISSUE #18: Rename all instances of AddressBook to FriendlyLink

### PR DESCRIPTION
**Pull request by Zxun2**
Fixes #18, #20 

- updated the naming where needed
- tests passed 248/248

### CODE REVIEW

**Comment by Zxun2** [comment]
CI will probably fail since there are a lot of check style violations in the main branch

**Comment by Zxun2** [comment]
Looks like i managed to fix it

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ private static int findPrefixPosition(String argsString, String prefix, int from
     private static ArgumentMultimap extractArguments(String argsString, List<PrefixPosition> prefixPositions) {
 
         // Sort by start position
-        prefixPositions.sort((prefix1, prefix2) -> prefix1.getStartPosition() - prefix2.getStartPosition());
+        prefixPositions.sort(Comparator.comparingInt(PrefixPosition::getStartPosition));
```
Comment (gohyongjing) [inline src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:90]: I like this as it makes the code cleaner

File: src/main/java/seedu/address/storage/JsonFriendlyLinkStorage.java [hunk 2]
```diff
@@ -12,42 +12,42 @@
...
-    public Optional<ReadOnlyAddressBook> readAddressBook(Path filePath) throws DataConversionException {
+    public Optional<ReadOnlyFriendlyLink> readFriendlyLink(Path filePath) throws DataConversionException {
         requireNonNull(filePath);
 
-        Optional<JsonSerializableAddressBook> jsonAddressBook = JsonUtil.readJsonFile(
-                filePath, JsonSerializableAddressBook.class);
-        if (!jsonAddressBook.isPresent()) {
+        Optional<JsonSerializableFriendlyLink> jsonAddressBook = JsonUtil.readJsonFile(
+                filePath, JsonSerializableFriendlyLink.class);
+        if (jsonAddressBook.isEmpty()) {
```
Comment (gohyongjing) [inline src/main/java/seedu/address/storage/JsonFriendlyLinkStorage.java:50]: I like this part where you changed not isPresent() to isEmpty(), which makes the code cleaner

**Review by gohyongjing — COMMENTED** [review]
LGTM

**Pull request merged**